Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -264,6 +264,18 @@ class BrowserViewController: UIViewController {
             if result == .walletCreated {
                 self.rewards.ledger.setupDeviceCheckEnrollment(client) { }
             }
+            
+            let isRewardsPanelShowing = (self.presentedViewController as? PopoverController)?
+                .contentController is RewardsPanelController
+            
+            // We do not want to show first ad when Rewards panel is opened. This happens when user
+            // enabled Rewards for the first time in the panel. In that case the ad will show after
+            // the panel is closed.
+            // This call is used if Rewards are enabled from another location, for example when user
+            // enables Rewards in one of NTP notifications.
+            if !isRewardsPanelShowing {
```

Comment:
This also solves another case when user enables rewards in the panel and immediately closes it